src/components/Product/Product.jsx

```diff
@@ -0,0 +1,156 @@
+            className="input search-bar"
+            onChange={(e) => searchProduct(e.target.value)}
+            placeholder="Search products"
+          />
+        </div>
+      </div>
+      <div className="product__filter--wrapper">
+        <div
+          className={`${
+            showFilter ? "product__filter--mobile" : "product__filter"
+          } `}
+        >
+          <div className="filter__heading">
+            <button
+              onClick={() => setShowFilter(!showFilter)}
+              className="button button--outline button--sm subtitle--sm"
+            >
+              {showFilter ? "Apply" : "Filters"}
+            </button>
+            <button
+              className="button button--primary text--white button--sm subtitle--sm"
+              onClick={clearAllFilters}
+            >
+              Clear All
+            </button>
+          </div>
+          <div className="filter__section">
+            <fieldset className="fieldset--style spacing--vh">
+              <legend> Price Sort by </legend>
+              <div>
+                <input
+                  onChange={sortPriceLowToHigh}
+                  checked={sortBy && sortBy === Actions.PRICE_LOW_TO_HIGH}
+                  type="radio"
+                  name="sort"
+                />
+                <label> Low to High </label>
+              </div>
+              <div>
+                <input
+                  onChange={sortPriceHighToLow}
+                  type="radio"
+                  name="sort"
+                  checked={sortBy && sortBy === Actions.PRICE_HIGH_TO_LOW}
+                />
+                <label> High to Low </label>
+              </div>
+            </fieldset>
+            <fieldset className="fieldset--style spacing--vh">
+              <legend> Availability </legend>
+              <div>
+                <input onChange={filterOutOfStock} type="checkbox" />
+                <label> Include Out of Stock </label>
+              </div>
+              <div>
+                <input onChange={filterFastDelivery} type="checkbox" />
+                <label> Fast Delivery </label>
+              </div>
+            </fieldset>
+          </div>
+        </div>
+        <div className={`${showFilter ? "hide" : "filter__panel--heading"}`}>
+          <button
+            className="button button--outline button--sm subtitle--sm"
+            onClick={() => setShowFilter(!showFilter)}
+          >
+            Filters
+          </button>
+          <button
+            className="button button--primary text--white button--sm subtitle--sm"
+            onClick={clearAllFilters}
+          >
+            Clear All
+          </button>
+        </div>
+        {searchedData.length > 0 ? (
+          <ProductListing productList={searchedData} />
+        ) : (
+          search && (
+            <div className="h6 text--center"> No items found with {search}</div>
+          )
+        )}
+      </div>
+    </div>
+  );
+}
```

Comment:
This file has code that can be broken down into smaller components. Again not something mandatory, but will make your code look even more cleaner.